packages/dara-core/dara/core/internal/encoder_registry.py

```diff
@@ -99,6 +100,7 @@ def _get_pandas_array_encoder(array_type: Type[Any], dtype: Any, raise_: bool =
     pandas.Index: Encoder(serialize=lambda x: x.to_list(), deserialize=lambda x: pandas.Index(x)),
     pandas.Timestamp: Encoder(serialize=lambda x: x.isoformat(), deserialize=lambda x: pandas.Timestamp(x)),
     pandas.DataFrame: Encoder(
-        serialize=lambda x: jsonable_encoder(x.to_dict(orient='dict')), deserialize=lambda x: _not_implemented(x,pandas.DataFrame)
-    )
+        serialize=lambda x: jsonable_encoder(x.to_dict(orient='dict')),
+        deserialize=lambda x: _not_implemented(x, pandas.DataFrame),
```

Comment:
Actually, could you test how this behaves with DataVariables? i.e. you should still be able to annotate an input as DataFrame for data variables:

```python
def resolver(x: pandas.DataFrame):
  foo

data_var = DataVariable(...)

DerivedVariable(resolver, variables=[data_var])
```

This might attempt to run through the deserialize logic? e.g. the deserializer should work if `x` is already a DataFrame. Otherwise it should fail as we can't know exactly how to process a dict into a DataFrame, preserving the types correctly